server/service/users.go

```diff
@@ -409,8 +410,11 @@ func (svc *Service) ModifyUser(ctx context.Context, userID uint, p fleet.UserPay
 		return nil, err
 	}
 
-	// load user again to get team-details like names.
-	user, err = svc.User(ctx, userID)
+	// Load user again to get team-details like names.
+	// Since we just modified the user and the changes may not have replicated to the read replica(s) yet,
+	// we must use the master to ensure we get the most up-to-date information.
+	ctxUsePrimary := ctxdb.RequirePrimary(ctx, true)
+	user, err = svc.User(ctxUsePrimary, userID)
```

Comment:
Another approach is for svc.ds.SaveUser itself (and `svc.saveUser(ctx, user)`) to return the updated user.